### ISSUE TITLE
farmhash fails to compile native extensions on a Raspberry Pi3

### ISSUE THREAD

**Issue by davestephens** [lovell/farmhash#12]
During an ````npm install```` on a Raspberry Pi3, Farmhash native extensions don't compile and the install fails. Interestingly, it works fine on a Raspberry Pi1.

Note that when compiling on an RPi1 and copying over to an RPi3, it works fine - it's just the actual compile that fails.

````
make: Entering directory '/home/bbs/enigma-bbs/node_modules/farmhash/build'
  CXX(target) Release/obj.target/farmhash-legacy/src/upstream/farmhash-legacy.o
*** Error in `g++': double free or corruption (top): 0x014a3c48 ***
farmhash-legacy.target.mk:103: recipe for target 'Release/obj.target/farmhash-legacy/src/upstream/farmhash-legacy.o' failed
make: *** [Release/obj.target/farmhash-legacy/src/upstream/farmhash-legacy.o] Aborted
make: Leaving directory '/home/bbs/enigma-bbs/node_modules/farmhash/build'

gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/bbs/.nvm/versions/node/v4.4.7/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Linux 4.4.11-v7+
gyp ERR! command "/home/bbs/.nvm/versions/node/v4.4.7/bin/node" "/home/bbs/.nvm/versions/node/v4.4.7/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/bbs/enigma-bbs/node_modules/farmhash
gyp ERR! node -v v4.4.7
gyp ERR! node-gyp -v v3.3.1
gyp ERR! not ok 
npm ERR! Linux 4.4.11-v7+
npm ERR! argv "/home/bbs/.nvm/versions/node/v4.4.7/bin/node" "/home/bbs/.nvm/versions/node/v4.4.7/bin/npm" "install"
npm ERR! node v4.4.7
npm ERR! npm  v2.15.8
npm ERR! code ELIFECYCLE

npm ERR! farmhash@1.2.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
````

***Potentially Useful Info***
**RPi3 (doesn't work):**
````node --version````: v4.4.7
````uname -a````: Linux raspberrypi 4.4.11-v7+ #888 SMP Mon May 23 20:10:33 BST 2016 armv7l GNU/Linux
````gcc --version````: gcc (Raspbian 4.9.2-10) 4.9.2

**RPi1 (does work):**
````uname -a````: Linux raspberrypi 4.1.19+ #858 Tue Mar 15 15:52:03 GMT 2016 armv6l GNU/Linux
````gcc --version````: gcc (Raspbian 4.9.2-10) 4.9.2

**Comment by lovell** [comment]
My best guess would be you've run into a gcc bug specific to the ARMv7 instruction set available on the Pi3. When you compile on the Pi1, it will generate ARMv6 instructions.

https://gcc.gnu.org/bugzilla/show_bug.cgi?id=70132 seems a likely candidate.

Perhaps try something like the following to workaround this:
```sh
CFLAGS="-march=armv7-a" CXXFLAGS="-march=armv7-a" npm install farmhash
```

**Comment by davestephens** [comment]
Yes, seems to be that. Annoyingly looks like it's been fixed for ages, but not backported into Raspbian Jessie. Given it's not a farmhash specific issue you might as well close this one off. 

Thanks for your help :+1:

**Comment by lovell** [comment]
Thanks for confirming. I'd prefer this to work on a Raspberry Pi 3, even with the current version of gcc in Raspbian, so consider this a bug.

We can probably make the `-march=native` flag conditional on the target architecture. Leave it with me, although I may need some help testing as I've only got ARMv6 Pi Zero to hand at the moment.

**Comment by lovell** [comment]
Commit 0bef353 on the `master` branch should fix this. Are you able to test locally?

Here's what I see locally on an x64 machine:

```sh
$ npm install --verbose 2>/dev/null | grep native | wc -l
4
$ npm install --arch=arm --verbose 2>/dev/null | grep native | wc -l
0
```

**Comment by davestephens** [comment]
Sure, happy to help test.

**Comment by davestephens** [comment]
Looks good!

This is an RPi3:
````
npm info it worked if it ends with ok
npm verb cli [ '/home/bbs/.nvm/versions/node/v4.4.7/bin/node',
npm verb cli   '/home/bbs/.nvm/versions/node/v4.4.7/bin/npm',
npm verb cli   'install',
npm verb cli   '--verbose' ]
npm info using npm@2.15.8
npm info using node@v4.4.7
npm verb readDependencies loading dependencies from /home/bbs/dev/farmhash/package.json
npm verb install where, deps [ '/home/bbs/dev/farmhash', [ 'nan' ] ]
npm verb install where, peers [ '/home/bbs/dev/farmhash', [] ]
npm verb installManyTop reading for lifecycle /home/bbs/dev/farmhash/package.json
npm info preinstall farmhash@1.2.1
npm verb installManyTop reading scoped package data from /home/bbs/dev/farmhash/node_modules/nan/package.json
npm verb readDependencies loading dependencies from /home/bbs/dev/farmhash/package.json
npm verb targetResolver reading package data from /home/bbs/dev/farmhash/node_modules/nan/package.json
npm verb already installed skipping nan@^2.4.0 /home/bbs/dev/farmhash
npm info build /home/bbs/dev/farmhash
npm info linkStuff farmhash@1.2.1
npm verb linkBins farmhash@1.2.1
npm verb linkMans farmhash@1.2.1
npm verb rebuildBundles farmhash@1.2.1
npm verb rebuildBundles [ 'nan' ]
npm info install farmhash@1.2.1

> farmhash@1.2.1 install /home/bbs/dev/farmhash
> node-gyp rebuild

gyp info it worked if it ends with ok
gyp verb cli [ '/home/bbs/.nvm/versions/node/v4.4.7/bin/node',
gyp verb cli   '/home/bbs/.nvm/versions/node/v4.4.7/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js',
gyp verb cli   'rebuild' ]
gyp info using node-gyp@3.3.1
gyp info using node@4.4.7 | linux | arm
gyp verb command rebuild []
gyp verb command clean []
gyp verb clean removing "build" directory
gyp verb command configure []
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp verb check python checking for Python executable "python2" in the PATH
gyp verb `which` succeeded python2 /usr/bin/python2
gyp verb check python version `/usr/bin/python2 -c "import platform; print(platform.python_version());"` returned: "2.7.9\n"
gyp verb get node dir no --target version specified, falling back to host node version: 4.4.7
gyp verb command install [ '4.4.7' ]
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp verb install input version string "4.4.7"
gyp verb install installing version: 4.4.7
gyp verb install --ensure was passed, so won't reinstall if already installed
gyp verb install version is already installed, need to check "installVersion"
gyp verb got "installVersion" 9
gyp verb needs "installVersion" 9
gyp verb install version is good
gyp verb get node dir target node version installed: 4.4.7
gyp verb build dir attempting to create "build" dir: /home/bbs/dev/farmhash/build
gyp verb build dir "build" dir needed to be created? /home/bbs/dev/farmhash/build
gyp verb build/config.gypi creating config file
gyp verb build/config.gypi writing out config file: /home/bbs/dev/farmhash/build/config.gypi
gyp verb config.gypi checking for gypi file: /home/bbs/dev/farmhash/config.gypi
gyp verb common.gypi checking for gypi file: /home/bbs/dev/farmhash/common.gypi
gyp verb gyp gyp format was not specified; forcing "make"
gyp info spawn /usr/bin/python2
gyp info spawn args [ '/home/bbs/.nvm/versions/node/v4.4.7/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/home/bbs/dev/farmhash/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/home/bbs/.nvm/versions/node/v4.4.7/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/home/bbs/.node-gyp/4.4.7/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/home/bbs/.node-gyp/4.4.7',
gyp info spawn args   '-Dnode_gyp_dir=/home/bbs/.nvm/versions/node/v4.4.7/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=node.lib',
gyp info spawn args   '-Dmodule_root_dir=/home/bbs/dev/farmhash',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp verb command build []
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp verb build type Release
gyp verb architecture arm
gyp verb node dev dir /home/bbs/.node-gyp/4.4.7
gyp verb `which` succeeded for `make` /usr/bin/make
make: Entering directory '/home/bbs/dev/farmhash/build'
  g++ '-DNODE_GYP_MODULE_NAME=farmhash-legacy' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DFARMHASH_LEGACY' '-DBUILDING_NODE_EXTENSION' -I/home/bbs/.node-gyp/4.4.7/include/node -I/home/bbs/.node-gyp/4.4.7/src -I/home/bbs/.node-gyp/4.4.7/deps/uv/include -I/home/bbs/.node-gyp/4.4.7/deps/v8/include -I../node_modules/nan  -fPIC -pthread -Wall -Wextra -Wno-unused-parameter -O3 -ffunction-sections -fdata-sections -fno-omit-frame-pointer -fno-rtti -fno-exceptions -std=gnu++0x -fexceptions -Wall -Ofast -flto -funroll-loops -MMD -MF ./Release/.deps/Release/obj.target/farmhash-legacy/src/upstream/farmhash-legacy.o.d.raw   -c -o Release/obj.target/farmhash-legacy/src/upstream/farmhash-legacy.o ../src/upstream/farmhash-legacy.cc
gyp info spawn make
gyp info spawn args [ 'V=1', 'BUILDTYPE=Release', '-C', 'build' ]
  g++ '-DNODE_GYP_MODULE_NAME=farmhash-legacy' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DFARMHASH_LEGACY' '-DBUILDING_NODE_EXTENSION' -I/home/bbs/.node-gyp/4.4.7/include/node -I/home/bbs/.node-gyp/4.4.7/src -I/home/bbs/.node-gyp/4.4.7/deps/uv/include -I/home/bbs/.node-gyp/4.4.7/deps/v8/include -I../node_modules/nan  -fPIC -pthread -Wall -Wextra -Wno-unused-parameter -O3 -ffunction-sections -fdata-sections -fno-omit-frame-pointer -fno-rtti -fno-exceptions -std=gnu++0x -fexceptions -Wall -Ofast -flto -funroll-loops -MMD -MF ./Release/.deps/Release/obj.target/farmhash-legacy/src/bindings.o.d.raw   -c -o Release/obj.target/farmhash-legacy/src/bindings.o ../src/bindings.cc
  g++ -shared -pthread -rdynamic  -Wl,-soname=farmhash-legacy.node -o Release/obj.target/farmhash-legacy.node -Wl,--start-group Release/obj.target/farmhash-legacy/src/upstream/farmhash-legacy.o Release/obj.target/farmhash-legacy/src/bindings.o -Wl,--end-group 
  rm -rf "Release/farmhash-legacy.node" && cp -af "Release/obj.target/farmhash-legacy.node" "Release/farmhash-legacy.node"
  g++ '-DNODE_GYP_MODULE_NAME=farmhash' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DBUILDING_NODE_EXTENSION' -I/home/bbs/.node-gyp/4.4.7/include/node -I/home/bbs/.node-gyp/4.4.7/src -I/home/bbs/.node-gyp/4.4.7/deps/uv/include -I/home/bbs/.node-gyp/4.4.7/deps/v8/include -I../node_modules/nan  -fPIC -pthread -Wall -Wextra -Wno-unused-parameter -O3 -ffunction-sections -fdata-sections -fno-omit-frame-pointer -fno-rtti -fno-exceptions -std=gnu++0x -fexceptions -Wall -Ofast -flto -funroll-loops -MMD -MF ./Release/.deps/Release/obj.target/farmhash/src/upstream/farmhash.o.d.raw   -c -o Release/obj.target/farmhash/src/upstream/farmhash.o ../src/upstream/farmhash.cc
  g++ '-DNODE_GYP_MODULE_NAME=farmhash' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DBUILDING_NODE_EXTENSION' -I/home/bbs/.node-gyp/4.4.7/include/node -I/home/bbs/.node-gyp/4.4.7/src -I/home/bbs/.node-gyp/4.4.7/deps/uv/include -I/home/bbs/.node-gyp/4.4.7/deps/v8/include -I../node_modules/nan  -fPIC -pthread -Wall -Wextra -Wno-unused-parameter -O3 -ffunction-sections -fdata-sections -fno-omit-frame-pointer -fno-rtti -fno-exceptions -std=gnu++0x -fexceptions -Wall -Ofast -flto -funroll-loops -MMD -MF ./Release/.deps/Release/obj.target/farmhash/src/bindings.o.d.raw   -c -o Release/obj.target/farmhash/src/bindings.o ../src/bindings.cc
  g++ -shared -pthread -rdynamic  -Wl,-soname=farmhash.node -o Release/obj.target/farmhash.node -Wl,--start-group Release/obj.target/farmhash/src/upstream/farmhash.o Release/obj.target/farmhash/src/bindings.o -Wl,--end-group 
  rm -rf "Release/farmhash.node" && cp -af "Release/obj.target/farmhash.node" "Release/farmhash.node"
make: Leaving directory '/home/bbs/dev/farmhash/build'
gyp info ok 
npm verb unsafe-perm in lifecycle true
npm info postinstall farmhash@1.2.1
npm info prepublish farmhash@1.2.1
npm verb validateInstall loading /home/bbs/dev/farmhash/package.json for validation
npm verb exit [ 0, true ]
npm info ok 
````

...and and RPi1 for completeness:
````
npm info it worked if it ends with ok
npm verb cli [ '/home/pi/.nvm/versions/node/v4.4.0/bin/node',
npm verb cli   '/home/pi/.nvm/versions/node/v4.4.0/bin/npm',
npm verb cli   'install',
npm verb cli   '--verbose' ]
npm info using npm@2.14.20
npm info using node@v4.4.0
npm verb readDependencies loading dependencies from /home/pi/dev/farmhash/package.json
npm verb install where, deps [ '/home/pi/dev/farmhash', [ 'nan' ] ]
npm verb install where, peers [ '/home/pi/dev/farmhash', [] ]
npm verb installManyTop reading for lifecycle /home/pi/dev/farmhash/package.json
npm info preinstall farmhash@1.2.1
npm verb readDependencies loading dependencies from /home/pi/dev/farmhash/package.json
npm verb cache add spec nan@^2.4.0
npm verb addNamed ">=2.4.0 <3.0.0" is a valid semver range for nan
npm verb addNameRange registry:https://registry.npmjs.org/nan not in flight; fetching
npm verb request uri https://registry.npmjs.org/nan
npm verb request no auth needed
npm info attempt registry request try #1 at 4:02:20 PM
npm verb request id d1748a5f458a8b74
npm verb etag "1OOZJAUMBQC59OINHZGMRRCSL"
npm http request GET https://registry.npmjs.org/nan
npm http 304 https://registry.npmjs.org/nan
npm verb headers { date: 'Sat, 03 Sep 2016 16:02:21 GMT',
npm verb headers   via: '1.1 varnish',
npm verb headers   'cache-control': 'max-age=300',
npm verb headers   etag: '"1OOZJAUMBQC59OINHZGMRRCSL"',
npm verb headers   age: '71',
npm verb headers   connection: 'keep-alive',
npm verb headers   'x-served-by': 'cache-lhr6342-LHR',
npm verb headers   'x-cache': 'HIT',
npm verb headers   'x-cache-hits': '3',
npm verb headers   'x-timer': 'S1472918541.157855,VS0,VE0',
npm verb headers   vary: 'Accept-Encoding' }
npm verb etag https://registry.npmjs.org/nan from cache
npm verb get saving nan to /home/pi/.npm/registry.npmjs.org/nan/.cache.json
npm verb correctMkdir /home/pi/.npm correctMkdir not in flight; initializing
npm verb addNamed "2.4.0" is a plain semver version for nan
npm verb afterAdd /home/pi/.npm/nan/2.4.0/package/package.json not in flight; writing
npm verb correctMkdir /home/pi/.npm correctMkdir not in flight; initializing
npm verb afterAdd /home/pi/.npm/nan/2.4.0/package/package.json written
npm info install nan@2.4.0 into /home/pi/dev/farmhash
npm info installOne nan@2.4.0
npm verb installOne of nan to /home/pi/dev/farmhash not in flight; installing
npm verb correctMkdir /home/pi/.npm/_locks correctMkdir not in flight; initializing
npm verb lock using /home/pi/.npm/_locks/nan-b4c2efdbb0fabd02.lock for /home/pi/dev/farmhash/node_modules/nan
npm verb unbuild node_modules/nan
npm verb gentlyRm don't care about contents; nuking /home/pi/dev/farmhash/node_modules/nan
npm verb tar unpack /home/pi/.npm/nan/2.4.0/package.tgz
npm verb tar unpacking to /home/pi/dev/farmhash/node_modules/nan
npm verb gentlyRm don't care about contents; nuking /home/pi/dev/farmhash/node_modules/nan
npm verb write writing to /home/pi/dev/farmhash/node_modules/nan/package.json
npm info preinstall nan@2.4.0
npm verb readDependencies loading dependencies from /home/pi/dev/farmhash/node_modules/nan/package.json
npm verb readDependencies loading dependencies from /home/pi/dev/farmhash/node_modules/nan/package.json
npm verb about to build /home/pi/dev/farmhash/node_modules/nan
npm info build /home/pi/dev/farmhash/node_modules/nan
npm info linkStuff nan@2.4.0
npm verb linkBins nan@2.4.0
npm verb linkMans nan@2.4.0
npm verb rebuildBundles nan@2.4.0
npm info install nan@2.4.0
npm info postinstall nan@2.4.0
npm verb unlock done using /home/pi/.npm/_locks/nan-b4c2efdbb0fabd02.lock for /home/pi/dev/farmhash/node_modules/nan
npm info build /home/pi/dev/farmhash
npm info linkStuff farmhash@1.2.1
npm verb linkBins farmhash@1.2.1
npm verb linkMans farmhash@1.2.1
npm verb rebuildBundles farmhash@1.2.1
npm verb rebuildBundles [ 'nan' ]
npm info install farmhash@1.2.1

> farmhash@1.2.1 install /home/pi/dev/farmhash
> node-gyp rebuild

gyp info it worked if it ends with ok
gyp verb cli [ '/home/pi/.nvm/versions/node/v4.4.0/bin/node',
gyp verb cli   '/home/pi/.nvm/versions/node/v4.4.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js',
gyp verb cli   'rebuild' ]
gyp info using node-gyp@3.3.0
gyp info using node@4.4.0 | linux | arm
gyp verb command rebuild []
gyp verb command clean []
gyp verb clean removing "build" directory
gyp verb command configure []
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp verb check python checking for Python executable "python2" in the PATH
gyp verb `which` succeeded python2 /usr/bin/python2
gyp verb check python version `/usr/bin/python2 -c "import platform; print(platform.python_version());"` returned: "2.7.9\n"
gyp verb get node dir no --target version specified, falling back to host node version: 4.4.0
gyp verb command install [ '4.4.0' ]
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp verb install input version string "4.4.0"
gyp verb install installing version: 4.4.0
gyp verb install --ensure was passed, so won't reinstall if already installed
gyp verb install version is already installed, need to check "installVersion"
gyp verb got "installVersion" 9
gyp verb needs "installVersion" 9
gyp verb install version is good
gyp verb get node dir target node version installed: 4.4.0
gyp verb build dir attempting to create "build" dir: /home/pi/dev/farmhash/build
gyp verb build dir "build" dir needed to be created? /home/pi/dev/farmhash/build
gyp verb build/config.gypi creating config file
gyp verb build/config.gypi writing out config file: /home/pi/dev/farmhash/build/config.gypi
gyp verb config.gypi checking for gypi file: /home/pi/dev/farmhash/config.gypi
gyp verb common.gypi checking for gypi file: /home/pi/dev/farmhash/common.gypi
gyp verb gyp gyp format was not specified; forcing "make"
gyp info spawn /usr/bin/python2
gyp info spawn args [ '/home/pi/.nvm/versions/node/v4.4.0/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/home/pi/dev/farmhash/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/home/pi/.nvm/versions/node/v4.4.0/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/home/pi/.node-gyp/4.4.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/home/pi/.node-gyp/4.4.0',
gyp info spawn args   '-Dnode_gyp_dir=/home/pi/.nvm/versions/node/v4.4.0/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=node.lib',
gyp info spawn args   '-Dmodule_root_dir=/home/pi/dev/farmhash',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp verb command build []
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp verb build type Release
gyp verb architecture arm
gyp verb node dev dir /home/pi/.node-gyp/4.4.0
gyp verb `which` succeeded for `make` /usr/bin/make
gyp info spawn make
gyp info spawn args [ 'V=1', 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/home/pi/dev/farmhash/build'
  g++ '-DNODE_GYP_MODULE_NAME=farmhash-legacy' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DFARMHASH_LEGACY' '-DBUILDING_NODE_EXTENSION' -I/home/pi/.node-gyp/4.4.0/include/node -I/home/pi/.node-gyp/4.4.0/src -I/home/pi/.node-gyp/4.4.0/deps/uv/include -I/home/pi/.node-gyp/4.4.0/deps/v8/include -I../node_modules/nan  -fPIC -pthread -Wall -Wextra -Wno-unused-parameter -O3 -ffunction-sections -fdata-sections -fno-omit-frame-pointer -fno-rtti -fno-exceptions -std=gnu++0x -fexceptions -Wall -Ofast -flto -funroll-loops -MMD -MF ./Release/.deps/Release/obj.target/farmhash-legacy/src/upstream/farmhash-legacy.o.d.raw   -c -o Release/obj.target/farmhash-legacy/src/upstream/farmhash-legacy.o ../src/upstream/farmhash-legacy.cc
  g++ '-DNODE_GYP_MODULE_NAME=farmhash-legacy' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DFARMHASH_LEGACY' '-DBUILDING_NODE_EXTENSION' -I/home/pi/.node-gyp/4.4.0/include/node -I/home/pi/.node-gyp/4.4.0/src -I/home/pi/.node-gyp/4.4.0/deps/uv/include -I/home/pi/.node-gyp/4.4.0/deps/v8/include -I../node_modules/nan  -fPIC -pthread -Wall -Wextra -Wno-unused-parameter -O3 -ffunction-sections -fdata-sections -fno-omit-frame-pointer -fno-rtti -fno-exceptions -std=gnu++0x -fexceptions -Wall -Ofast -flto -funroll-loops -MMD -MF ./Release/.deps/Release/obj.target/farmhash-legacy/src/bindings.o.d.raw   -c -o Release/obj.target/farmhash-legacy/src/bindings.o ../src/bindings.cc
  g++ -shared -pthread -rdynamic  -Wl,-soname=farmhash-legacy.node -o Release/obj.target/farmhash-legacy.node -Wl,--start-group Release/obj.target/farmhash-legacy/src/upstream/farmhash-legacy.o Release/obj.target/farmhash-legacy/src/bindings.o -Wl,--end-group 
  rm -rf "Release/farmhash-legacy.node" && cp -af "Release/obj.target/farmhash-legacy.node" "Release/farmhash-legacy.node"
  g++ '-DNODE_GYP_MODULE_NAME=farmhash' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DBUILDING_NODE_EXTENSION' -I/home/pi/.node-gyp/4.4.0/include/node -I/home/pi/.node-gyp/4.4.0/src -I/home/pi/.node-gyp/4.4.0/deps/uv/include -I/home/pi/.node-gyp/4.4.0/deps/v8/include -I../node_modules/nan  -fPIC -pthread -Wall -Wextra -Wno-unused-parameter -O3 -ffunction-sections -fdata-sections -fno-omit-frame-pointer -fno-rtti -fno-exceptions -std=gnu++0x -fexceptions -Wall -Ofast -flto -funroll-loops -MMD -MF ./Release/.deps/Release/obj.target/farmhash/src/upstream/farmhash.o.d.raw   -c -o Release/obj.target/farmhash/src/upstream/farmhash.o ../src/upstream/farmhash.cc
  g++ '-DNODE_GYP_MODULE_NAME=farmhash' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DBUILDING_NODE_EXTENSION' -I/home/pi/.node-gyp/4.4.0/include/node -I/home/pi/.node-gyp/4.4.0/src -I/home/pi/.node-gyp/4.4.0/deps/uv/include -I/home/pi/.node-gyp/4.4.0/deps/v8/include -I../node_modules/nan  -fPIC -pthread -Wall -Wextra -Wno-unused-parameter -O3 -ffunction-sections -fdata-sections -fno-omit-frame-pointer -fno-rtti -fno-exceptions -std=gnu++0x -fexceptions -Wall -Ofast -flto -funroll-loops -MMD -MF ./Release/.deps/Release/obj.target/farmhash/src/bindings.o.d.raw   -c -o Release/obj.target/farmhash/src/bindings.o ../src/bindings.cc
  g++ -shared -pthread -rdynamic  -Wl,-soname=farmhash.node -o Release/obj.target/farmhash.node -Wl,--start-group Release/obj.target/farmhash/src/upstream/farmhash.o Release/obj.target/farmhash/src/bindings.o -Wl,--end-group 
  rm -rf "Release/farmhash.node" && cp -af "Release/obj.target/farmhash.node" "Release/farmhash.node"
make: Leaving directory '/home/pi/dev/farmhash/build'
gyp info ok 
npm verb unsafe-perm in lifecycle true
npm info postinstall farmhash@1.2.1
npm info prepublish farmhash@1.2.1
npm verb validateInstall loading /home/pi/dev/farmhash/package.json for validation
nan@2.4.0 node_modules/nan
npm verb exit [ 0, true ]
npm info ok 

````

**Comment by lovell** [comment]
Great, this fix is available in v1.2.1 now available via npm. Thank you for reporting this problem.

**Comment by davestephens** [comment]
:+1:
